File: python/bit_manipulation/checkSetBit.py
def check():
    query = input()
    while int(query):
        num = input().split(" ")
        f = 1
        f = f << int(num[1])
        res = int(num[0]) & f

        if res == 0:
            print("False")
        else:
            print("True")
        query = int(query) - 1


if __name__ == "__main__":
    check()

File: cpp/bit_manipulation/checkSetBit.cpp
/*
check ith bit is set of not
*/

#include <bits/stdc++.h>
using namespace std;

int main()
{
    cout << "Enter number of queries: " << endl;
    int query; 
    cin >> query;

    int n, i;

    while (query--)
    {

        cin >> n >> i;
        int f = 1;
        f = f << i;

        int res = n & f;

        if (res == 0)
        {
            cout << "False" << endl;
        }
        else
        {
            cout << "True" << endl;
        }
    }
}